src/Messages.jl

```diff
@@ -119,7 +119,7 @@ Response(body) = Response(200; body=body)
 Base.convert(::Type{Response}, s::AbstractString) = Response(s)
 
 function reset!(r::Response)
-    r.version = v"1.1"
+    r.version = HTTPVersion(1, 1)
```

Comment:
Could construct the version number without going through a string instead maybe?

Reply:
This case is actually fine, we could leave this as-is. 

The problematic case is `parse_status_line!`/`parse_request_line!` where we're parsing the version out of the response/request body